mesh/mesh.cpp

```diff
@@ -5957,12 +5982,14 @@ void Mesh::UniformRefinement2D()
             AverageVertices(vv, 2, oedge+e[ei]);
          }
 
-         elements[j+0] = new Triangle(oedge+e[1], oedge+e[2], oedge+e[0], attr);
-         elements[j+1] = new Triangle(oedge+e[0], v[1], oedge+e[1], attr);
-         elements[j+2] = new Triangle(oedge+e[2], oedge+e[1], v[2], attr);
-
-         v[1] = oedge+e[0];
-         v[2] = oedge+e[2];
+         new_elements[j++] =
+            new Triangle(oedge+e[2], v[0], oedge+e[0], attr);
```

Comment:
The order of the vertices here is changed from master -- we want to keep the marked edge from the parent. This is in case the uniform refinement is followed by local refinement.

Reply:
Also, the embedding matrix will be different.